### ISSUE TITLE
vmalert:  Add the Bearer Token authorization option

### ISSUE THREAD

**Issue by Denissa89** [VictoriaMetrics/VictoriaMetrics#1608]
Hello VictoriaMetrics developers!
We use vmauth as auth proxy, router and load balancer for VictoriaMetrics cluster.
In vmaut we use the Bearer Token, so that our metrics sources can write data into VictoriaMetrics.
I noticed that vmalert cannot authenticate to vmauth, using Bearer Token.

So I want to create a feature request to implement this functionality.

Thank you!

**Comment by valyala** [comment]
`vmalert` gained support for Bearer token authorization starting from the commit 68721f6e7d6a9f3d9c680c4485a9b4647ff68811 . This commit adds the following command-line flags for configuring Bearer auth token:

```
  -datasource.bearerToken string
       Optional bearer auth token to use for -datasource.url.
  -datasource.bearerTokenFile string
       Optional path to bearer token file to use for -datasource.url.
  -remoteRead.bearerToken string
       Optional bearer auth token to use for -remoteRead.url.
  -remoteRead.bearerTokenFile string
       Optional path to bearer token file to use for -remoteRead.url.
  -remoteWrite.bearerToken string
       Optional bearer auth token to use for -remoteWrite.url.
  -remoteWrite.bearerTokenFile string
       Optional path to bearer token file to use for -remoteWrite.url.
```

Additionally, the commit adds the ability to read Basic auth password from file with the following options:

```
  -datasource.basicAuth.passwordFile string
       Optional path to basic auth password to use for -datasource.url
  -remoteRead.basicAuth.passwordFile string
       Optional path to basic auth password to use for -remoteRead.url
  -remoteWrite.basicAuth.passwordFile string
       Optional path to basic auth password to use for -remoteWrite.url
```

`vmalert` re-reads Bearer tokens and Basic auth passwords from files if the corresponding files change. This allows using dynamic update for auth tokens.

The commit will be included in the next release.

**Comment by valyala** [comment]
In the mean time it is possible to build vmalert form the commit 68721f6e7d6a9f3d9c680c4485a9b4647ff68811 according to [these instructions](https://docs.victoriametrics.com/vmalert.html#how-to-build-from-sources) and verify whether Bearer token authorization works as expected there.

**Comment by Denissa89** [comment]
thanks!

**Comment by valyala** [comment]
The [Bearer token authorization](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1608#issuecomment-919073087) is available starting from [vmalert v1.66.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as done.
